torch_geometric/data/batch.py

```diff
@@ -210,3 +305,98 @@ def __len__(self) -> int:
     def __reduce__(self):
         state = self.__dict__.copy()
         return DynamicInheritanceGetter(), self.__class__.__bases__, state
+
+    def add_node_attr(self, attrname: str, attr: Tensor) -> None:
+        r"""Adds an attribute to the nodes in an existing batch.
+        The first dimension of the :obj:`attr` must match the number
+        of nodes in the batch. The exisiting
+        :obj:`~torch_geometric.data.Batch.batch` will be used to
+        assign the elements to the correct graph.
+        """
+        assert attr.device == self.batch.device
+        batch_idxs = self.batch
+
+        self[attrname] = attr
+        out = batch_idxs.unique(return_counts=True)[1]
+        out = out.cumsum(dim=0)
+        self._slice_dict[attrname] = self._pad_zero(out).cpu()
+
+        self._inc_dict[attrname] = torch.zeros(self._num_graphs,
+                                               dtype=torch.long)
+
+    def add_graph_attr(self, attrname: str, attr: Tensor) -> None:
+        r"""Adds an attribute to the graphs in an existing batch.
+        The first dimension of the :obj:`attr` must match the
+        number of nodes in the batch. The exisiting :obj:`~Batch.batch`
+        will be used to assign the elements to the correct graph.
+        """
+        assert attr.device == self.batch.device
+
+        self[attrname] = attr
+        self._slice_dict[attrname] = torch.arange(self.num_graphs + 1,
+                                                  dtype=torch.long)
+
+        self._inc_dict[attrname] = torch.zeros(self.num_graphs,
+                                               dtype=torch.long)
+
+    def set_edge_index(self, edge_index: Tensor,
```

Comment:
This interface is a bit unclear to me. Wouldn't it make more sense to just expect `list[Tensor]` here? Also, what's the use-case for this?

Reply:
I guess it should be the new EdgeIndex Type.
I would like to be able to predict edges, update the graph and run it through a GNN again without losing the ability to slice the batch.

Reply:
I implemented both ways now. Passing the tensors directly should only really matter for very large batches.
I can also remove it and only implement the `List[Tensor]` method. 
What do you prefer?